src/components/LaatijaUpload/LaatijaUpload.svelte

```diff
@@ -0,0 +1,123 @@
+<script>
+  import * as R from 'ramda';
+
+  import * as Fetch from '@Utility/fetch-utils';
+  import * as Future from '@Utility/future-utils';
+
+  import { locale, _ } from '@Language/i18n';
+
+  import Fileupload from '@Component/Fileupload/Fileupload';
+  import Table from '@Component/Table/Table';
+  import Button from '@Component/Button/Button';
+
+  import { flashMessageStore } from '@/stores';
+
+  import {
+    parse,
+    validate,
+    readData,
+    dataValid,
+    postLaatijatFuture
+  } from './laatija-utils';
+  import Patevuustaso from './Patevyystaso';
+  import Date from './Date';
+
+  let laatijaData;
+  let update = data => {
+    // TODO: Same file selection does not trigger this
+    flashMessageStore.flush('Laatija');
+    let files = R.compose(data)({});
+    const reader = new FileReader();
+    reader.onload = e => {
+      laatijaData = e.target.result;
+    };
+    reader.readAsText(files.files[0], 'UTF-8');
+  };
```

Comment:
Wrapping this into Future that assigns laatijaData with Either-value would be more akin to what we've been doing in other places. No need to refac this now, but I might do it in the near future.